src/PactNet.Native/NativeMockServer.cs

```diff
@@ -103,5 +89,60 @@ public bool WithRequestBody(InteractionHandle interaction, string contentType, s
 
         public bool WithResponseBody(InteractionHandle interaction, string contentType, string body)
             => NativeInterop.WithBody(interaction, InteractionPart.Response, contentType, body);
+
+        #endregion Http Interaction Model Support
+
+        #region Message Interaction Model Support
+
+        public void WriteMessagePactFile(MessagePactHandle pact, string directory, bool overwrite)
+            => VerifyResult(() => NativeInterop.WriteMessagePactFile(pact, directory, overwrite));
+
+        public bool WithMessagePactMetadata(MessagePactHandle pact, string @namespace, string name, string value)
+            => NativeInterop.WithMessagePactMetadata(pact, @namespace, name, value);
+
+        public MessagePactHandle NewMessagePact(string consumerName, string providerName)
+            => NativeInterop.NewMessagePact(consumerName, providerName);
+
+        public MessageHandle NewMessage(MessagePactHandle pact, string description)
+            => NativeInterop.NewMessage(pact, description);
+
+        public bool ExpectsToReceive(MessageHandle message, string description)
+            => NativeInterop.MessageExpectsToReceive(message, description);
+        public bool Given(MessageHandle message, string description)
+            => NativeInterop.MessageGiven(message, description);
+
+        public bool GivenWithParam(MessageHandle message, string description, string name, string value)
+            => NativeInterop.MessageGivenWithParam(message, description, name, value);
+
+        public bool WithMetadata(MessageHandle message, string key, string value)
+            => NativeInterop.MessageWithMetadata(message, key, value);
+
+        public bool WithContents(MessageHandle message, string contentType, string body, uint size)
+            => NativeInterop.MessageWithContents(message, contentType, body, new UIntPtr(size));
+
+        public string Reify(MessageHandle message)
+        {
+            var allocatedString = Marshal.PtrToStringAnsi(NativeInterop.MessageReify(message));
+            return allocatedString;
+        }
+
+
+        #endregion Message Interaction Model Support
+
+        private void VerifyResult(Func<int> func)
```

Comment:
Are we certain that the request/response and message pact styles will always have the same response types/codes?